### PR TITLE
plan: return table dual directly when the condition is false.

### PR DESCRIPTION
**Pull request by hanfei1991**
@shenli @coocood @zimulala @winoros PTAL

### CODE REVIEW

File: plan/new_physical_plan_builder.go [hunk 1]
```diff
@@ -208,7 +208,26 @@ func (p *DataSource) getMemTableTask(prop *requiredProp) (task taskProfile, err
 	memTable.Ranges = rb.buildTableRanges(fullRange)
 	task = &rootTaskProfile{p: memTable}
 	task = prop.enforceProperty(task, p.ctx, p.allocator)
-	return task, p.storeTaskProfile(prop, task)
+	return task, nil
+}
+
+func (p *DataSource) getDualTask() (taskProfile, error) {
```
Comment (shenli) [inline plan/new_physical_plan_builder.go]: Add comment for this function. And tryToGetDualTask may be a better name.

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
@shenli fixed, and this change is only for new plan. it will not inflence tidb-test.

**Comment by shenli** [comment]
LGTM

**Pull request merged**